src/main/java/decodex/storage/Storage.java

```diff
@@ -107,8 +108,14 @@ public void loadRecipesFromDirectory(ModuleManager moduleManager, RecipeManager
      *
      * @return The list of recipe File objects.
      */
-    private File[] getAllRecipeFiles() {
+    private File[] getAllRecipeFiles() throws StorageException {
         File recipeDirectory = new File(DEFAULT_RECIPE_DIRECTORY);
+
+        if (!recipeDirectory.isDirectory()) {
+            // throw error if not a valid directory.
+            throw new StorageException(ErrorMessages.INVALID_RECIPE_DIRECTORY_FILETYPE);
+        }
+
         File[] files = recipeDirectory.listFiles();
         File[] recipeFiles = Arrays.stream(files)

```

Comment:
Perhaps only files with the `.txt` extension should be loaded to prevent reading of unsupported file types.